files/BeverControl/BeverControl.ifc

```diff
@@ -200,23 +200,36 @@ DATA;
-#187= IFCALIGNMENT('2AZWily2L0CR6uYzpQefac',#10,'My alignment',$,$,#189,#190,$);
+#187= IFCALIGNMENT('2eUtJU0J1AwP6L_1ZRV4jS',#10,'My alignment',$,$,#189,#190,$);
 #189= IFCLOCALPLACEMENT(#3,#2);
 #190= IFCPRODUCTDEFINITIONSHAPE($,$,(#183,#186));
-#191= IFCRELNESTS('1RZFUgXNmtc2RDiSok0MPI',#10,$,$,#187,(#71,#87));
+#191= IFCRELNESTS('00iDnuT2OmZ5EX33iyPCve',#10,$,$,#187,(#71,#87));
 #192= IFCPROPERTYSINGLEVALUE('Station',$,IFCLENGTHMEASURE(0.0),$);
-#193= IFCPROPERTYSET('2QQikaaO55PeJVnvtHH87N',#10,'Pset_Stationing',$,(#192));
-#194= IFCREFERENT('2AOoo7uD113RNsK0jik7N0',#10,'Alignment start',$,$,#198,$,.STATION.);
-#195= IFCRELNESTS('1kBkCaNLYj5C2XqwYgJKAl',#10,$,$,#187,(#194));
+#193= IFCPROPERTYSET('0PqyDdvjD25wC76$0o4u0s',#10,'Pset_Stationing',$,(#192));
+#194= IFCREFERENT('0l7g75I6n13O4mx71lNFf3',#10,'Alignment start',$,$,#198,$,.STATION.);
+#195= IFCRELNESTS('1Q8EEKuLfismigK2Xy_f$L',#10,$,$,#187,(#194));
 #196= IFCPOINTBYDISTANCEEXPRESSION(IFCNONNEGATIVELENGTHMEASURE(0.0),$,$,$,#137);
 #197= IFCAXIS2PLACEMENTLINEAR(#196,$,$);
 #198= IFCLINEARPLACEMENT(#189,#197,$);
-#199= IFCRELPOSITIONS('2FZg9_uZ18ZQ0aKmWO_gJ9',#10,$,$,#194,(#4));
-#200= IFCRELAGGREGATES('2xgsarNZDCWf7oXSY$JpYY',#10,$,$,#13,(#187));
-#201= IFCBUILTSYSTEM('38dH5f1hL8rOiXps256_DV',#10,'My system',$,$,.TUNNEL_SUPPORT.,$);
-#202= IFCRELREFERENCEDINSPATIALSTRUCTURE('32zB7tnsTAB9QbpII_27ur',#10,$,$,(#201),#11);
-#203= IFCARCHELEMENT('1WeS7eMYrAbv3JqRiZFDNY',#10,$,$,$,$,$,$,$);
-#204= IFCRELASSIGNSTOGROUP('2bNrxGywf1MgdqXiSsv2zy',#10,$,$,(#203),$,#201);
+#199= IFCRELPOSITIONS('3x3qyHFLbF0vYxVlaH9iNr',#10,$,$,#194,(#4));
+#200= IFCCARTESIANPOINT((0.0,0.0,0.0));
+#201= IFCAXIS2PLACEMENT3D(#200,$,$);
+#202= IFCLOCALPLACEMENT(#198,#201);
+#203= IFCRELAGGREGATES('2yku8na7qjVXHO9jh4RZFg',#10,$,$,#13,(#187));
+#204= IFCBUILTSYSTEM('0VsYBdpZDCafmqdh5yRQFT',#10,'My system',$,$,.TUNNEL_SUPPORT.,$);
+#205= IFCRELREFERENCEDINSPATIALSTRUCTURE('30EVEWa3HERO5Tvj_Q_a5M',#10,$,$,(#204),#11);
+#206= IFCARCHELEMENT('14NmFugJ54of$U96F0C9_z',#10,$,$,$,$,$,$,$);
+#207= IFCRELASSIGNSTOGROUP('2WsilcCkjFEhjxfK4Pov73',#10,$,$,(#206),$,#204);
+#208= IFCCLASSIFICATIONREFERENCE($,'My tunnel identification','My tunnel classification',#33,$,$);
+#209= IFCRELASSOCIATESCLASSIFICATION('1SjPqCepvCvBzK1WNbi3mP',#10,$,$,(#11),#208);
+#210= IFCLIBRARYREFERENCE('https://bsi-infraroom.github.io/IFC-Documentation-Tunnel/4_4_0_0/general/HTML/annex/annex-a/general-usage/IFC4x3.ifc','#3540','Pset_LandRegistration',$,$,#44);
+#211= IFCPROPERTYSET('3pDp2r8kD24xWlJ_DNOGXH',#10,'Pset_LandRegistration',$,(#213,#214,#215));
+#212= IFCRELDEFINESBYPROPERTIES('1vy2Ik$tP5owGTBSwD0pHv',#10,$,$,(#4),#211);
+#213= IFCPROPERTYSINGLEVALUE('LandId',$,IFCLABEL('47'),$);
+#214= IFCPROPERTYSINGLEVALUE('IsPermanentId',$,IFCBOOLEAN(.T.),$);
+#215= IFCPROPERTYSINGLEVALUE('LandTitleId',$,IFCLABEL('NO'),$);
+#216= IFCRELDEFINESBYPROPERTIES('2X91TmiPfAdvy8b_8673WU',#10,$,$,(#4),#211);
+#217= IFCVOXELGRID(1.0,2.0,3.0,1,1,1,(.T.,.T.,.T.));
```

Comment:
Looking at attributes 4.-6.:

- there is 1 voxel in x direction
- there is 1 voxel in y direction
- there is 1 voxel in z direction

resulting in the grid having 1x1x1 = 1 voxel.

However, looking at 7. attribute, there are 3 values provided. This seems off.

Reply:
@SergejMuhic @larswik Would this be a good case for a where rule?

Reply:
Please check the capital letters of "LandId", "IsPermanentId", "LandTitleId"?

Reply:
Apologies - I wasn't precise enough. My comment applies only to `#217 = IFCVOXELGRID ...`.

Reply:
Might not even have to be a where rule. Maybe just:
```
NumberOfVoxelsX : IfcInteger];
NumberOfVoxelsY : OPTIONAL IfcInteger;
NumberOfVoxelsZ : OPTIONAL IfcInteger;
Voxels : LIST [1 : (NumberOfVoxelsX * NumberOfVoxelsY * NumberOfVoxelsZ)] OF IfcBoolean;
```
Not sure whether the parenthesis are needed. Just in case.

Reply:
The expression would need to account for cardinality 'optional' of the attributes. This would make the expression somewhat more complicated. (IMO, a where rule would be an easier solution.)

Reply:
Yes, sure. I meant it more as an alternative to where rule since we can already initialize correctly. To me it would be a better way compared to verifying after it is initialized.
```
Voxels : LIST [1 : IfcVoxelSize(NumberOfVoxelsX, NumberOfVoxelsY, NumberOfVoxelsZ)] OF IfcBoolean;
FUNCTION IfcVoxelSize(NumberOfVoxelsX : IfcInteger, NumberOfVoxelsY : IfcInteger, NumberOfVoxelsZ : IfcInteger) : INTEGER;
  LOCAL
    NumberOfVoxels : INTEGER;      
  END_LOCAL:
  IF EXISTS(NumberOfVoxelsY) THEN
    NumberOfVoxels = NumberOfVoxelsX * NumberOfVoxelsY:
  ELSE
    NumberOfVoxels = NumberOfVoxelsX *NumberOfVoxelsX;
  END_IF;
  IF EXISTS(NumberOfVoxelsZ) THEN
    NumberOfVoxels = NumberOfVoxels * NumberOfVoxelsZ:
  ELSE
    NumberOfVoxels = NumberOfVoxels *NumberOfVoxelsX;
  END_IF;
END_FUNCTION;
```